lib/Bio/EnsEMBL/DataCheck/Checks/DescriptionNewlines.pm

```diff
@@ -0,0 +1,52 @@
+=head1 LICENSE
+
+Copyright [2018-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::DescriptionNewlines;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'DescriptionNewlines',
+  DESCRIPTION    => 'Check for newlines and tabs in gene descriptions',
+  GROUPS         => ['xref'],
+  DATACHECK_TYPE => 'critical',
+  TABLES         => ['gene']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_1 = 'gene description does not contain  newlines and/or tabs';
+  my $sql_1  = qq/
+  	SELECT COUNT(*) 
+        FROM gene 
```

Comment:
Collection databases necessitate some joins here. The issue is that this test will be run once per species - for standard core dbs that's fine, because 1 db = 1 species. A collection db can have up to 250 species; running the same query 250 times is wasteful, and if there is a problem, it's not immediately obvious which species are causing the problem.

The solution is to make the query species-specific, for which we need to join to the `coord_system` table, via `seq_region`, in order to apply the necessary condition. For example, https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/UniProtDisplayXref.pm#L48.

It's not applicable here, but some tables are not species-specific, in which case you can set the `-per_db` flag when creating the datacheck - this will ensure that for collection dbs, it only gets run once. E.g. https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/DuplicateXref.pm#L36

If you're wondering why the originating healthcheck didn't have the correct SQL, that's because healthchecks can sometimes pre-date collection dbs; in other cases they were just not written with collection dbs in mind.